### PR TITLE
9 filter sort pagination

### PR DESCRIPTION
**Pull request by PaulBredl**
## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?
Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
    
## Checklist before requesting a review
- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My code fulfilles all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,7 +14,7 @@ graphqlCodegen {
     // all config options:
     // https://github.com/kobylynskyi/graphql-java-codegen/blob/main/docs/codegen-options.md
     outputDir = new File("$buildDir/generated")
-    packageName = "de.unistuttgart.iste.gits.courseservice.dto"
+    packageName = "de.unistuttgart.iste.gits.generated.dto"
```
Comment (PaulBredl) [inline build.gradle:17]: Changes from template repo

File: docker-compose.yml [hunk 2]
```diff
@@ -13,6 +13,20 @@ services:
       - POSTGRES_USER=root
       - POSTGRES_PASSWORD=root
       - POSTGRES_DB=course_service
+  testDb:
```
Comment (PaulBredl) [inline docker-compose.yml:16]: Changes from template repo

File: src/main/java/de/unistuttgart/iste/gits/common/util/PaginationUtil.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package de.unistuttgart.iste.gits.common.util;
+
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/common/util/PaginationUtil.java:2]: Changes from template repo

File: src/main/java/de/unistuttgart/iste/gits/courseservice/persistence/dao/ChapterEntity.java [hunk 4]
```diff
@@ -36,10 +38,7 @@ public class ChapterEntity {
     @Column(nullable = false)
     private OffsetDateTime endDate;
 
-    @ManyToOne
-    private CourseEntity course;
-
-    @ElementCollection
-    private List<UUID> contentIds;
+    @Column(nullable = false, name = "course_id")
+    private UUID courseId;
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/courseservice/persistence/dao/ChapterEntity.java:42]: Only having the uuid is enough, in cases where we have the id of the course, like when creating a new chapter for a course, this means that we don't have to query the course from the db but only set the id

File: src/test/java/de/unistuttgart/iste/gits/util/ClearDatabase.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
+package de.unistuttgart.iste.gits.util;
```
Comment (PaulBredl) [inline src/test/java/de/unistuttgart/iste/gits/util/ClearDatabase.java:1]: this package is also from template repo

**Comment by PaulBredl** [comment]
Please squash on merge

**Review by myluki2000 — APPROVED** [review]
Looks good 👍
